tests/c/strarray.c

```diff
@@ -0,0 +1,54 @@
+// Run-time:
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_JITSTATE=1
+//   stderr:
+//     jit-state: start-tracing
+//     pepper
+//     jit-state: stop-tracing
+//     cucumber
+//     jit-state: enter-jit-code
+//     tomato
+//     jit-state: exit-jit-code
+//     jit-state: enter-jit-code
+//     banana
+//     jit-state: deoptimise
+//     ...
+//     jit-state: exit-jit-code
+//   stdout:
+//     exit
+
+// Check that basic trace compilation works.
```

Comment:
This comment isn't very informative. In fact, I'd suggest it's misleading! Isn't the point of this test the recursive global thingy? I wondered if it might sense to even do something like:

```
static char *x = "...";
static char *y = "...";
static char *[] = {x, y};
```

I'm not sure if that's valid C or not though!

Reply:
Oh forgot to update that description. Will fix.

Reply:
Fixed the comment in [ff92c79](https://github.com/ykjit/yk/pull/618/commits/ff92c79ca99ac236b53c21d2cf101b8ca17bb192). There's no need to actually create different globals. In LLVM IR a global array will automatically create a separate global for each element, and then put the pointers in the array instead.

Reply:
>  There's no need to actually create different globals.

I understand (or, at least, guessed!) that, but wouldn't it be clearer if we separated this out rather than relying on LLVM's (possibly fragile) internal behaviour?

Reply:
Also, the comment says "global arrays" but the PR is "recursive global variables". I assume the former is just a special case of the latter?

Reply:
The PR says "clone global variables recursively", i.e. look at each element in the global array and clone them too if they are global variables.

I could add a check to see that the AOT IR does indeed contain a global variable array with pointers to other globals.

Reply:
> The PR says "clone global variables recursively", i.e. look at each element in the global array

But notice you said "clone global *variables* recursively" but then global *array*. It's confusing for thickies like me!

Reply:
I'm talking about an LLVM `GlobalVariable` that is underneath a `ConstantArray`.

Reply:
Let me try another fix.

Reply:
Try if this works better for you: [26cd73c](https://github.com/ykjit/yk/pull/618/commits/26cd73c19abe075fc753c6b0ee448e24591c797a)